test/exe/global_deploy_test.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+require 'test_helper'
+require 'krane/cli/krane'
+require 'krane/global_deploy_task'
+
+class GlobalDeployTest < Krane::TestCase
+  def test_global_deploy_with_default_options
+    set_krane_global_deploy_expectations!
+    krane_global_deploy!
+  end
+
+  def test_deploy_parses_global_timeout
+    set_krane_global_deploy_expectations!(new_args: { global_timeout: 10 })
+    krane_global_deploy!(flags: '--global-timeout 10s')
+    set_krane_global_deploy_expectations!(new_args: { global_timeout: 60**2 })
+    krane_global_deploy!(flags: '--global-timeout 1h')
+  end
+
+  def test_deploy_passes_verify_result
+    set_krane_global_deploy_expectations!(run_args: { verify_result: true })
+    krane_global_deploy!(flags: '--verify-result true')
+    set_krane_global_deploy_expectations!(run_args: { verify_result: false })
+    krane_global_deploy!(flags: '--verify-result false')
+  end
+
+  def test_deploy_passes_filename
+    set_krane_global_deploy_expectations!(new_args: { filenames: ['/my/file/path'] })
+    krane_global_deploy!(flags: '-f /my/file/path')
+    set_krane_global_deploy_expectations!(new_args: { filenames: ['/my/other/file/path'] })
+    krane_global_deploy!(flags: '--filenames /my/other/file/path')
+  end
+
+  def test_deploy_parses_selector
+    selector = 'name:web'
+    set_krane_global_deploy_expectations!(new_args: { selector: selector })
+    krane_global_deploy!(flags: "--selector #{selector}")
+  end
+
+  private
+
+  def set_krane_global_deploy_expectations!(new_args: {}, run_args: {})
+    options = default_options(new_args, run_args)
+    selector_args = options[:new_args][:selector]
+    selector = mock('LabelSelector')
+    Krane::LabelSelector.expects(:parse).with(selector_args).returns(selector)
```

Comment:
Why go to all this effort to stub out the selector parsing? Your dummy selector values are invalid right now.

Reply:
The `with` on line 48 is really hard to write if you want to allow any instance of a class inside of a hash, but still enforce everything else.

Reply:
Did you try using an `instance_of` parameter matcher? https://mocha.jamesmead.org/Mocha/ParameterMatchers.html#instance_of-instance_method